--- a/modules/JSON-XS/JSON-XS/XS.pm
+++ b/modules/JSON-XS/JSON-XS/XS.pm
@@ -101,15 +101,13 @@

 package JSON::XS;

-use common::sense;
-
 our $VERSION = 3.04;
 our @ISA = qw(Exporter);

 our @EXPORT = qw(encode_json decode_json);

 use Exporter;
-use XSLoader;
+use XSLoader ();

 use Types::Serialiser ();

@@ -1744,7 +1742,7 @@
    *JSON::XS::Boolean:: = *Types::Serialiser::Boolean::;
 }

-XSLoader::load "JSON::XS", $VERSION;
+XSLoader::load("JSON::XS", $VERSION);

 =head1 SEE ALSO

@@ -1758,4 +1756,3 @@
 =cut

 1
-
